pkg/controller/migration/convert/network.go

```diff
@@ -119,6 +130,107 @@ func handleNetwork(c *components, install *Installation) error {
 }
 
+func handleNonCalicoCNI(c *components, install *Installation) error {
+	plugin, err := getCNIPlugin(c)
+	if err != nil {
+		return err
+	}
+	if plugin == operatorv1.PluginCalico {
+		return nil
+	}
+
+	context := fmt.Sprintf("detected %s CNI plugin", plugin)
+
+	icc := getContainer(c.node.Spec.Template.Spec, containerInstallCNI)
+	if icc != nil {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("%s, %s container is not supported in the node daemonset", context, containerInstallCNI),
+		}
+	}
+
+	// CALICO_NETWORKING_BACKEND
+	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return err
+	}
+	if netBackend == nil || *netBackend != "none" {
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, CALICO_NETWORKING_BACKEND=none is expected", context)}
+	}
+
+	if install.Spec.CNI == nil {
+		install.Spec.CNI = &operatorv1.CNISpec{}
+	}
+
+	switch plugin {
+	case operatorv1.PluginAmazonVPC:
+		install.Spec.CNI.Type = plugin
+		// FELIX_IPTABLESMANGLEALLOWACTION
+		mangleAllow, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_IPTABLESMANGLEALLOWACTION")
+		if err != nil {
+			return err
+		}
+
+		// check that IPTABLESMANGLE is set Return
+		if mangleAllow == nil {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESMANGLEALLOWACTION was unset, expected it to be 'Return'.", context)}
+		} else if *mangleAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESMANGLEALLOWACTION was %s, expected it to be 'Return'.", context, *mangleAllow)}
+		}
+	case operatorv1.PluginAzureVNET:
+		install.Spec.CNI.Type = plugin
+	case operatorv1.PluginGKE:
+		install.Spec.CNI.Type = plugin
+		// FELIX_IPTABLESMANGLEALLOWACTION
+		mangleAllow, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_IPTABLESMANGLEALLOWACTION")
+		if err != nil {
+			return err
+		}
+
+		// check that IPTABLESMANGLE is set Return
+		if mangleAllow == nil {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESMANGLEALLOWACTION was unset, expected it to be 'Return'.", context)}
+		} else if *mangleAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESMANGLEALLOWACTION was %s, expected it to be 'Return'.", context, *mangleAllow)}
+		}
+
+		// FELIX_IPTABLESFILTERALLOWACTION
+		filterAllow, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_IPTABLESFILTERALLOWACTION")
+		if err != nil {
+			return err
+		}
+		// check that IPTABLESFILTER is set Return
+		if filterAllow == nil {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESFILTERALLOWACTION was unset, expected it to be 'Return'.", context)}
+		} else if *filterAllow != "Return" {
+			return ErrIncompatibleCluster{fmt.Sprintf("%s, FELIX_IPTABLESFILTERALLOWACTION was set to %s, expected it to be 'Return'.", context, *filterAllow)}
+		}
+	default:
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unsupported plugin migrate: '%s'.", plugin),
```

Comment:
"unsupported plugin migrate" isn't really a complete thought. Maybe "unable to migrate plugin %s: unsupported"?